src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseGitDiffReportService.java

```diff
@@ -156,7 +156,7 @@ private ProgrammingExerciseFullGitDiffEntryDTO convertToFullEntry(ProgrammingExe
     public ProgrammingExerciseGitDiffReport updateReport(ProgrammingExercise programmingExercise) {
         // Synchronized to prevent multiple git-diffs to be generated for the same exercise at the same time
         // This happens e.g. when creating an exercise
-        synchronized (programmingExercise) {
+        synchronized (("ProgrammingExerciseGitDiffReport-" + programmingExercise.getId()).intern()) {
```

Comment:
I think we should find a different solution. Why can this even happen when creating an exercise? Is there a design flaw that this methods is invoked multiple times...

Reply:
This would happen when the build plans finished at the same time, which apparently leads to a scenario where this method runs twice. This would then lead to the case that they both deleted the old report at the same time and afterward saved the new report. This meant that no report got deleted, but both reports got saved.
I could not see any other check against it, so I used synchronized here.

I would normally assume that Hibernate handles these cases, as the relationship between ProgrammingExercise and ProgrammingExerciseGitDiffReport is a OneToOne, but this is not the case in Artemis.